Examples/Python/src/GeometryBuildingGen1.cpp

```diff
@@ -0,0 +1,155 @@
+
+    auto config =
+        py::class_<LayerCreator::Config>(creator, "Config").def(py::init<>());
+    ;
+
+    ACTS_PYTHON_STRUCT_BEGIN(config, LayerCreator::Config);
+    ACTS_PYTHON_MEMBER(surfaceArrayCreator);
+    ACTS_PYTHON_MEMBER(cylinderZtolerance);
+    ACTS_PYTHON_MEMBER(cylinderPhiTolerance);
+    ACTS_PYTHON_STRUCT_END();
+  }
+
+  {
+    using Creator = Acts::SurfaceArrayCreator;
+    using Config = typename Creator::Config;
+
+    auto creator =
+        py::class_<Creator, std::shared_ptr<Creator>>(m, "SurfaceArrayCreator")
+            .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  {
+    using Base = Acts::ILayerArrayCreator;
+    using Creator = Acts::LayerArrayCreator;
+    using Config = typename Creator::Config;
+
+    py::class_<Base, std::shared_ptr<Base>>(m, "ILayerArrayCreator");
+
+    auto creator = py::class_<Creator, std::shared_ptr<Creator>, Base>(
+                       m, "LayerArrayCreator")
+                       .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  {
+    using Base = Acts::ITrackingVolumeArrayCreator;
+    using Creator = Acts::TrackingVolumeArrayCreator;
+    using Config = typename Creator::Config;
+
+    py::class_<Base, std::shared_ptr<Base>>(m, "ITrackingVolumeArrayCreator");
+
+    auto creator = py::class_<Creator, std::shared_ptr<Creator>, Base>(
+                       m, "TrackingVolumeArrayCreator")
+                       .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  py::class_<Acts::VolumeBounds, std::shared_ptr<Acts::VolumeBounds>>(
+      m, "VolumeBounds");
+
+  py::class_<Acts::CylinderVolumeBounds,
+             std::shared_ptr<Acts::CylinderVolumeBounds>, Acts::VolumeBounds>(
+      m, "CylinderVolumeBounds")
+      .def(py::init<ActsScalar, ActsScalar, ActsScalar, ActsScalar, ActsScalar,
+                    ActsScalar, ActsScalar>(),
+           "rmin"_a, "rmax"_a, "halfz"_a, "halfphi"_a = M_PI, "avgphi"_a = 0.,
+           "bevelMinZ"_a = 0., "bevelMaxZ"_a = 0.);
+
+  {
+    // NOTE Here we break the interface down from a transform to a zshift, since
+    // this is the only thing used
+    auto helper =
+        py::class_<Acts::CylinderVolumeHelper>(m, "CylinderVolumeHelper")
+            .def(py::init([](const Acts::CylinderVolumeHelper::Config &cfg,
+                             Acts::Logging::Level level) {
+              return Acts::CylinderVolumeHelper(
+                  cfg, Acts::getDefaultLogger("CylinderVolumeHelper", level));
+            }))
+            .def("createTrackingVolume",
+                 [](const Acts::CylinderVolumeHelper &self,
+                    GeometryContext gctx, const LayerVector &layers,
+                    std::shared_ptr<VolumeBounds> volumeBounds,
+                    ActsScalar zShift, const std::string &name) {
+                   auto trafo = Transform3::Identity();
```

Comment:
If there's not good way to create the transform in Python without adding a bunch of API from Eigen, maybe this could be a separate Python helper, and this function still accepts the transform and doesn't convert internally?

Reply:
as above, I would stick to the "make ITk work" type of bindings...

Reply:
Sure, but this is just confusing. I don't think we can and should just throw all other considerations over board for "making it work".

Reply:
I changed it to take a `Transform3`.